resource/plugins/unreal/python/opener/finalizers/unreal_asset_opener_finalizer.py

```diff
@@ -0,0 +1,171 @@
+# :coding: utf-8
+# :copyright: Copyright (c) 2014-2022 ftrack
+import json
+import os.path
+
+import unreal
+
+import ftrack_api
+
+from ftrack_connect_pipeline import constants as core_constants
+from ftrack_connect_pipeline.utils import str_version
+from ftrack_connect_pipeline_unreal import plugin
+from ftrack_connect_pipeline_unreal.utils import (
+    custom_commands as unreal_utils,
+)
+from ftrack_connect_pipeline_unreal.constants import asset as asset_const
+from ftrack_connect_pipeline_unreal.asset import UnrealFtrackObjectManager
+from ftrack_connect_pipeline_unreal.asset.dcc_object import UnrealDccObject
+
+
+class UnrealAssetOpenerFinalizerPlugin(plugin.UnrealOpenerFinalizerPlugin):
+    '''Unreal asset open finalizer plugin.'''
+
+    plugin_name = 'unreal_asset_opener_finalizer'
+
+    def run(self, context_data=None, data=None, options=None):
+        '''Restore asset info.'''
+
+        result = {}
+
+        # First, evaluate asset path
+        asset_filesystem_path = None
+        for comp in data:
+            if comp['type'] == core_constants.COMPONENT:
+                for result in comp['result']:
+                    if result['type'] == core_constants.IMPORTER:
+                        plugin_result = result['result'][0]
+                        asset_filesystem_path = list(
+                            plugin_result['result'].values()
+                        )[0]
+                        break
+
+        if asset_filesystem_path is None:
+            # No asset path found, abort
+            return {'message': 'No opened asset path found.'}
+
+        # Expect: "C:\\Users\\Henrik Norin\\Documents\\Unreal Projects\\MyEmptyProject\\Content\\Assets\\NewWorld.umap"
+        # Transform to asset path
+        asset_path = unreal_utils.filesystem_asset_path_to_asset_path(
+            asset_filesystem_path
+        )
+
+        # Load the asset in Unreal
+        self.logger.debug(
+            'Result of loading asset "{}" in Unreal editor: {}'.format(
+                asset_path, unreal.EditorAssetLibrary.load_asset(asset_path)
+            )
+        )
+
+        # Restore pipeline asset info
+        assetversion = self.session.query(
+            'AssetVersion where id is "{}"'.format(context_data['version_id'])
+        ).one()
+        ident = str_version(assetversion, by_task=False)
+        metadata = None
+        if 'ftrack-connect-pipeline-unreal' in assetversion['metadata']:
```

Comment:
Maybe this could be in utils as it is used in multiple places, something like get_asset_info_metadata

Reply:
I'll move to utils function